src/oop/special_class_method/medium/q11/MainTest.java

```diff
@@ -0,0 +1,266 @@
+                    double.class
+            };
+            assertTrue(testClass.hasMethod(setAttributeMethodName0, methodModifierClasses, Void.TYPE),
+                    String.format(modifiedClassMessage, setAttributeMethodName0));
+            testClass.callMethod(setAttributeMethodName0, setMethodArguments, testInstance);
+            assertTrue(testClass.hasModifier(setAttributeMethodName0, methodModifierClasses, "public"),
+                    String.format(modifiedClassMessage, setAttributeMethodName0));
+
+            _assertEquals(updatedValue, testClass.getFieldValue(testInstance, attributeName0),
+                    String.format(modifiedClassMessage, setAttributeMethodName0));
+        }
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(3.2, 3.4, "Red"),
+                Arguments.of(432.3, 10.2, "Black"),
+                Arguments.of(1023.4, 11.0, "Blue"),
+                Arguments.of(changeLimit, 3.4, "Red"),
+                Arguments.of(changeLimit - 0.01, 10.2, "Black"),
+                Arguments.of(changeLimit + 0.01, 11.0, "Blue"),
+                Arguments.of(434.3, 43.4, "White"),
+                Arguments.of(43.34, 7, "Yellow"),
+                Arguments.of(100, 7, "Yellow")
+        );
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctToStringMethod(double value1, double value2, String value3) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, double.class},
+                {value3, String.class}
+        };
+        String incorrectToStringMessage = "Your toString method does not return the correct String.";
+        Object testInstance = testClass.createInstance(arguments);
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod("toString", null), String.format(incorrectMethodDefinition, "toString", className));
+        assertTrue(testClass.hasModifier("toString", null, "public"), String.format(incorrectModifierMessage, "toString", className));
+        assertTrue(testClass.hasReturnType("toString", null, String.class), String.format(incorrectReturnType, "toString", className));
+        String expectedOutput = "The " + value3 + " Oven with size of " + value2 + " has the tempeture of " + value1;
+        Object toStringOutput = testClass.callMethod("toString", testInstance);
+        _assertEquals(expectedOutput, toStringOutput, incorrectToStringMessage);
+    }
+
+    @Test
+    public void powerDecreaseIsDefinedCorrectly() {
```

Comment:
I don't think we need the "isDefinedCorrectly" methods since we test them in the following tests, but check with Seth